connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/importer/ProcessDefinitionSearch.java

```diff
@@ -65,4 +73,91 @@ public void query(Consumer<List<ProcessDefinition>> resultHandler) {
 
     } while (result.getItems().size() > 0);
   }
+
+  /**
+   * Fetches a list of process instances associated with a given process definition key, including
+   * the variables for each instance. The variables represent the current state of each process
+   * instance and may be updated over the lifetime of the instance.
+   *
+   * @param processDefinitionKey The unique identifier for the process definition to retrieve
+   *     instances of.
+   * @return A list of {@link ProcessInstance} objects, each representing a process instance and its
+   *     associated variables.
+   * @throws RuntimeException If an error occurs during the fetch operation.
+   */
+  public List<ProcessInstance> fetchProcessInstancesWithVariables(final Long processDefinitionKey) {
```

Comment:
I anticipate performance degradation for these two methods. Consider adding a "time consumed" metric (not in this PR, maybe in a separate task). We need to understand how much time on avg will it take to build this model.